src/Cryptol/TypeCheck/TypeMap.hs

```diff
@@ -131,23 +130,24 @@ instance TrieMap TypeMap Type where
       TUser _ _ t -> lookupTM t
       TVar x      -> lookupTM x . tvar
       TCon c ts   -> lookupTM ts <=< lookupTM c . tcon
-      TRec fs     -> let (xs,ts) = unzip $ sortBy (comparing fst) fs
+      TRec fs     -> let (xs,ts) = unzip $ canonicalFields fs
                      in lookupTM ts <=< lookupTM xs . trec
 
   alterTM ty f m =
     case ty of
       TUser _ _ t -> alterTM t f m
       TVar x      -> m { tvar = alterTM x f (tvar m) }
       TCon c ts   -> m { tcon = alterTM c (updSub ts f) (tcon m) }
-      TRec fs     -> let (xs,ts) = unzip $ sortBy (comparing fst) fs
+      TRec fs     -> let (xs,ts) = unzip $ canonicalFields fs
                      in m { trec = alterTM xs (updSub ts f) (trec m) }
 
   toListTM m =
     [ (TVar x,           v) | (x,v)   <- toListTM (tvar m) ] ++
     [ (TCon c ts,        v) | (c,m1)  <- toListTM (tcon m)
                             , (ts,v)  <- toListTM m1 ] ++
-    [ (TRec (zip fs ts), v) | (fs,m1) <- toListTM (trec m)
-                            , (ts,v)  <- toListTM m1 ]
+    [ (TRec (recordFromFields (zip fs ts)), v)
```

Comment:
Passing through this `TypeMap` seems like it will lose display order information.  I don't really understand how this datastructure is used, so I'm not sure if this is a problem.

Reply:
It seems that `TypesMap` is only used for a couple of Cryptol-to-Cryptol transformations in the `MonoValues` and `Specialize` modules. Losing the display order information for these is probably not a big deal, but it might be nice to point that out in a comment in `TypeMap.hs`, just in case anyone wants to fix it later.